framework/src/play-integration-test/src/test/scala/play/it/api/SecretConfigurationParserSpec.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.it.api
+
+import ch.qos.logback.classic.spi.ILoggingEvent
+import play.api.{ Environment, Mode }
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.PlaySpecification
+import play.it.LogTester
+
+import scala.util.Try
+
+class SecretConfigurationParserSpec extends PlaySpecification {
+
+  def parseSecret(mode: Mode)(extraConfig: (String, String)*): (Option[String], Seq[ILoggingEvent]) = {
+    Try {
+      val app = GuiceApplicationBuilder(environment = Environment.simple(mode = mode))
+        .configure(extraConfig: _*)
+        .build()
+      val (secret, events) = LogTester.recordLogEvents {
+        app.httpConfiguration.secret.secret
+      }
+      (secret, events)
+    } match {
+      case scala.util.Success((secret, events)) => (Option(secret), events)
+      case scala.util.Failure(_) => (None, Seq.empty)
+    }
+  }
+
+  "When parsing SecretConfiguration" should {
+    "in DEV mode" should {
+      "return 'changeme' when it is configured to it" in {
+        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> "changeme")
+        events.map(_.getFormattedMessage).find(_.contains("Generated dev mode secret")) must beSome
+        secret must beSome
+      }
+
+      "generate a secret when no value is configured" in {
+        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> null)
+        events.map(_.getFormattedMessage).find(_.contains("Generated dev mode secret")) must beSome
+        secret must beSome
+      }
+
+      "log a warning when secret length is smaller than 8 chars" in {
+        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> "1234567")
+        events.map(_.getFormattedMessage).find(_.contains("The application secret is too short and does not have the recommended amount of entropy")) must beSome
+        secret must beSome
+      }
+
+      "log a warning when secret length is smaller then 15 chars" in {
+        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> "1234567890123")
```

Comment:
`secret` is 13 chars long. Should be 14 to test the edge case.
```suggestion
        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> "12345678901234")
```